File: emp_backend/src/main/java/com/example/demo/model/NotificationSeenBy.java
package com.example.demo.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.*;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Builder
@Entity(name = "notification_seen_by")
public class NotificationSeenBy {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "notification_id")
    @JsonBackReference
    private Notification notification;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "employee_id")
    private Employee employee;
    private Boolean isSeen;
}

File: emp_frontend/src/app/models/notification-seen-by.ts
import {Employee} from "./employee";

export class NotificationSeenBy {
  id: number;
  notification!: Notification; // Reference to Notification model
  employee!: Employee; // Reference to Employee model
  isSeen!: boolean;


  constructor() {
    this.id = 0;
    //this.notification = notification;
    //this.employee = "";
    //this.isSeen = "";
  }
}
